### ISSUE TITLE
Version 2.0.0.RC1: CAS problem

### ISSUE THREAD

**Issue by tcrespog** [alvarosanchez/grails-spring-security-rest#379]
The plugin throws an exception when it's configured to use CAS authentication at the moment the _\<baseUrl\>/oauth/authenticate/cas_ URL is visited. The stacktrace is:

```
Caused by: org.pac4j.core.exception.TechnicalException: loginUrl, prefixUrl and restUrl cannot be all blank
	at org.pac4j.cas.config.CasConfiguration.internalInit(CasConfiguration.java:83)
	at org.pac4j.core.util.InitializableWebObject.init(InitializableWebObject.java:24)
	at org.pac4j.cas.client.CasClient.clientInit(CasClient.java:47)
	at org.pac4j.core.client.IndirectClient.internalInit(IndirectClient.java:49)
	at org.pac4j.core.util.InitializableWebObject.init(InitializableWebObject.java:24)
	at org.pac4j.core.client.IndirectClient.getRedirectAction(IndirectClient.java:83)
	at grails.plugin.springsecurity.rest.RestOauthController.authenticate(RestOauthController.groovy:72)
```
This problem doesn't happen in version 2.0.0.M2 of the plugin which relies on `org.pac4j:pac4j-cas:1.8.3` while RC1 relies on `org.pac4j:pac4j-cas:2.2.1`.

**Comment by tcrespog** [comment]
In a further investigation I have realized that the source of the problem consists on the way `org.pac4j.cas.client.CasClient` is configured between versions [1.8.3](https://github.com/pac4j/pac4j/blob/056a2bfa7cf04ccb64b7f9afa47c05d707283c83/pac4j-cas/src/main/java/org/pac4j/cas/client/CasClient.java#L101) and [2.2.1](https://github.com/pac4j/pac4j/blob/c0217f144f095c3e7d9cd1536beb39940a9d637a/pac4j-cas/src/main/java/org/pac4j/cas/client/CasClient.java#L35).

The plugin instantiates the clients injecting config values on their class properties ([source](https://github.com/alvarosanchez/grails-spring-security-rest/blob/75d026f928b30e9618dd5e099991b8eaf0902a01/spring-security-rest/grails-app/services/grails/plugin/springsecurity/rest/RestOauthService.groovy#L54)). On `org.pac4j:pac4j-cas:2.2.1` the configuration is no longer stored in properties of the class itself, but on a helper class: [`org.pac4j.cas.config.CasConfiguration`](https://github.com/pac4j/pac4j/blob/c0217f144f095c3e7d9cd1536beb39940a9d637a/pac4j-cas/src/main/java/org/pac4j/cas/config/CasConfiguration.java#L36).

**Comment by tcrespog** [comment]
In order to config the CAS client properly, the `casLoginUrl` config must be stated as:

```
grails.plugin.springsecurity.rest.oauth.cas.configuration = new org.pac4j.cas.config.CasConfiguration("https://my.cas-server.com/cas/login")
```
I may make a PR to reflect the change in the doc.

**Comment by alvarosanchez** [comment]
Fixed via ffa848c
